rq/queue.py

```diff
@@ -291,11 +305,13 @@ def compact(self):
     def push_job_id(self, job_id: str, pipeline: t.Optional['Pipeline'] = None, at_front=False):
         """Pushes a job ID on the corresponding Redis queue.
         'at_front' allows you to push the job onto the front instead of the back of the queue"""
+        self.log.debug(f"Pushing job {blue(job_id)} into the queue {green(self.name)}, current job count {self.count}")
```

Comment:
`self.count` incurs a Redis operation so I'd rather not have job count displayed here.

Reply:
We could have a toggle for "debug mode" to avoid the extra request while not debugging, what do you think? Without the count we would lose the ability to check whether the list was actually increased as it should.

Reply:
Introduced a `debug` param to the queue's `__init__`, and then logged the count only if the debug is `True` (defaults to `False`). This will avoid extra operations when not debugging while maintaining full visibility over the most relevant operations.

Reply:
Undone.